### PR TITLE
[scalardl-ledger] Support TLS in ScalarDL Ledger chart

### PR DESCRIPTION
**Pull request by kota2and3kan**
## Description

This PR adds TLS support in the ScalarDL Ledger chart.

In this update, users can configure TLS features and set key/cert files for ScalarDL Ledger.

Note: You need to enable the wire encryption feature on the ScalarDL Ledger side by setting `scalar.dl.ledger.server.tls.enabled=true` in the `ledger.ledgerProperties` in your custom values file.

```console
[Client] ---> [Envoy] ---(TLS)---> [ScalarDL Ledger]
```

## Related issues and/or PRs

* https://github.com/scalar-labs/scalar-envoy/pull/28
  * I updated the Scalar Envoy to support TLS both downstream and upstream connections.
* https://github.com/scalar-labs/helm-charts/pull/253
  * I updated the Helm Chart of Scalar Envoy to support TLS between Envoy and ScalarDL Ledger.
* https://github.com/scalar-labs/docs-internal-orchestration/pull/14
  * I updated the document that is related to this PR. You can see the usage of this new feature and the getting started guide.

## Changes made

* Add new values to configure the TLS configurations.
* Mount key/cert file to use TLS connections between client (Envoy) and ScalarDL Ledger in the Deployment manifest.

## Checklist

- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes.
- [x] Any remaining open issues linked to this PR are documented and up-to-date (Jira, GitHub, etc.).
- [x] Tests (unit, integration, etc.) have been added for the changes.
- [x] My changes generate no new warnings.
- [x] Any dependent changes in other PRs have been merged and published.

## Additional notes (optional)

N/A

## Release notes

Support TLS configuration in the ScalarDL Ledger chart. You can enable TLS between Envoy and ScalarDL Ledger.

### CODE REVIEW

**Review by supl — APPROVED** [review]
LGTM.
Thank you 🙇

**Review by choplin — APPROVED** [review]
LGTM!

File: charts/scalardl/templates/ledger/deployment.yaml [hunk 1]
```diff
@@ -114,13 +144,27 @@ spec:
               command:
               - /usr/local/bin/grpc_health_probe
               - -addr=:50051
+              {{- if .Values.ledger.tls.enabled }}
+              - -tls
+              {{- if .Values.ledger.tls.caRootCertSecret }}
+              - -tls-ca-cert=/tls/certs/ca-root-cert.pem
+              {{- end }}
+              - -tls-server-name={{ .Values.ledger.tls.overrideAuthority }}
```
Comment (feeblefakie) [inline charts/scalardl/templates/ledger/deployment.yaml:152]: Ditto. (The same question as the ScalarDB Cluster PR.)

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Left one question for clarification.

File: charts/scalardl/templates/ledger/deployment.yaml (hunk 1 above)
Comment (kota2and3kan) [reply charts/scalardl/templates/ledger/deployment.yaml:153]: Thank you for your review!
I answered it on the ScalarDB Cluster PR side.
https://github.com/scalar-labs/helm-charts/pull/255#discussion_r1535123060

**Review by jnmt — APPROVED** [review]
LGTM! But can you check why CI fails, just in case? Thank you!

**Pull request merged**